<a href="https://colab.research.google.com/github/ndhend/Winc_assignments/blob/main/Income_inequality_Winc.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#header
Income inequality in relation to GPD

#section
Choosing Between concat and merge:

If you simply want to stack DataFrames together, especially when they have the same columns, then pd.concat is appropriate. If you need to combine DataFrames based on common values in specific columns, then pd.merge is the better choice.

In [53]:
#import codes

import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

url_gdp = 'https://raw.githubusercontent.com/ndhend/csv_data_winc/main/gdp-per-capita-penn-world-table.csv'
url_ineq = 'https://raw.githubusercontent.com/ndhend/csv_data_winc/main/inequality.csv'
df_gdp = pd.read_csv(url_gdp)
df_inequality = pd.read_csv(url_ineq)
# display(df_gdp.head(50))
# display(df_inequality.head(50))

# rename column 'Entity' so it matches with the column in the df_inequality dataframe
df_gdp1 = df_gdp.rename(columns={'Entity': 'Country'})
# display(df_gdp1)

# merge both dataframes on the same column 'Country' so I have on df with gdp and ineq rates.
df_gdp_ineq = pd.merge(df_gdp1, df_inequality, on='Country')
# display(df_gdp_ineq.head(50))
# df_gdp_ineq.columns

# Assuming df_gdp_ineq is your original DataFrame
# names_to_drop = ['p60p70_thr_posttax_nat', 'p70p80_thr_posttax_nat', 'p80p90_thr_posttax_nat',
#                  'p90p100_thr_posttax_nat', 'p99_999p100_thr_posttax_nat', 'p99_99p100_thr_posttax_nat',
#                  'p99_9p100_thr_posttax_nat', 'p99p100_thr_posttax_nat', 'median_pretax', 'median_posttax_nat']

# df_gdp_ineq1 = df_gdp_ineq.drop(columns=names_to_drop)

# df_gdp_ineq1.columns
# names_to_drop1 = ['p99_999p100_thr_pretax', 'p99_99p100_thr_pretax',
#        'p99_9p100_thr_pretax', 'p99p100_thr_pretax', 'p0p10_thr_posttax_nat',
#        'p10p20_thr_posttax_nat', 'p20p30_thr_posttax_nat',
#        'p30p40_thr_posttax_nat', 'p40p50_thr_posttax_nat',
#        'p50p60_thr_posttax_nat']
# df_gdp_ineq2 = df_gdp_ineq1.drop(columns=names_to_drop1)
# df_gdp_ineq2.columns
gdp_ineq = df_gdp_ineq.rename(columns={'GDP per capita (output, multiple price benchmarks)': 'GDP_per_capita', 'Gini coefficient (before tax) (World Inequality Database)': 'Gini_index' })
gdp_ineq.squeeze()
# display(gdp_ineq.head(50))
gdp_gini = gdp_ineq[[ 'GDP_per_capita', 'Gini_index']].dropna()
display(gdp_gini)
gdp_gini.corr()



GDP_per_capita  Gini_index
16           3114.0884    0.467630
22           3114.0884    0.481698
25           3114.0884    0.474482
28           3114.0884    0.472132
32           3114.0884    0.467515
...                ...         ...
458141       2787.6590    0.624443
458146       2787.6590    0.624443
458161       2787.6590    0.624443
458167       2787.6590    0.635799
458169       2787.6590    0.685154

[168355 rows x 2 columns]

GDP_per_capita  Gini_index
GDP_per_capita        1.000000   -0.314347
Gini_index           -0.314347    1.000000

#Answer
The correlation coefficient of the GDP and Gini index is -0.314347.
-1 = perfect negative
+1 = perfect positive

Using the rule of thumb sizes from Rea and Parker(2014) the relationship is between 0.20 < 0.40 which they interprete as a, in this case, moderate negative relationship between gdp and gini_index, meaning that when one goes up, the other goes down or vice versa.